api/main.go

```diff
@@ -125,32 +195,37 @@ func setupProducer(brokerAddress []string) error {
 	producerConfig.Producer.Return.Successes = true
 	producerConfig.Producer.Return.Errors = true
 
-	producer, err := sarama.NewSyncProducer(producerBrokers, producerConfig)
+	log.Traceln("\tCreating sync producer...")
+	newProducer, err := sarama.NewSyncProducer(producerBrokers, producerConfig)
 	if err != nil {
 		return err
 	}
 
-	Producer = producer
+	*producer = newProducer
+	log.Traceln("\tSync producer has been created")
 
 	return nil
 }
 
-func setupConsumer(brokerAddress []string) error {
+func setupConsumer(brokerAddress []string, incomingTopics []string, consumer **cluster.Consumer) error {
```

Comment:
Can consumer argument be just a pointer *cluster.Consumer ?

Reply:
I have changed those global variable from pointer to struct to interface, so this problem is no more efficient, it will be replased in the next commit